src/PotentialTypes/EmpiricalPotentials/coulomb.jl

```diff
@@ -1,49 +1,41 @@
 ##############################   Coulomb  ###################################
-# mutable struct Coulomb <: EmpiricalPotential
-#     q_1::Float64
-#     q_2::Float64
-#     ϵ0::Float64
-# end
-
-# function Coulomb()
-#     #ToDO
-#     return Coulomb(1.0, 1.0, 1.0)
-# end
-
-# function get_trainable_params(c::Coulomb)
-#     p = Parameter{}(())
-#     return p
-# end
-
-# function get_nontrainable_params(c::Coulomb)
-#     return (q_1 = c.q_1, q_2 = c.q_2, ϵ0 = c.ϵ0)
-# end
+mutable struct Coulomb <: EmpiricalPotential
+    q_1::AbstractFloat
+    q_2::AbstractFloat
+    ϵ0::AbstractFloat
+    rcutoff::AbstractFloat
+end
+
+get_trainable_params(c::Coulomb) = Parameter{}(())
+
+get_nontrainable_params(c::Coulomb) = Parameter{:q1, :q2, :ϵ0, :rcutoff}((c.q_1, c.q_2, c.ϵ0, c.rcutoff))
 
 # ############################# Energies ##########################################
 
-# function potential_energy(r::Vector{<:Real}, p::Coulomb)
-#     return p.q_1 * p.q_2 / (4.0 * π * p.ϵ0 * norm(r))
-# end
+function potential_energy(R::AbstractFloat, c::Coulomb)
+    return c.q_1 * c.q_2 / (4.0 * π * c.ϵ0 * R)
+end
 
 # ############################### Forces ##########################################
 
-# function force(r::Vector{<:Real}, p::Coulomb)
-#     d = norm(r)
-#     return p.q_1 * p.q_2 / (4.0 * π * p.ϵ0 * d^2) .* r ./ d
-# end
+force(r::SVector{3,<:AbstractFloat}, c::Coulomb) = force(r, norm(r), c)
+
+function force(R::AbstractFloat, r::SVector{3,<:AbstractFloat}, c::Coulomb)
+    SVector(c.q_1 * c.q_2 / (4.0 * π * c.ϵ0 * R^2) .* r ./ R)
+end
 
 # ##############################   Gradients  ###################################
 
-# function grad_potential_energy(r::Vector{<:Real}, p::Coulomb)
-#     println("The Coulomb potential has no trainable parameters")
-#     return (dpdnull = 0.0)
-# end
-# function grad_force(r:: Vector{<:Real}, p::Coulomb)
-#     println("The Coulomb potential has no trainable parameters")
-#     return (dfdnull =  0 .* r ./ d )
-# end
-
-# function grad_virial(r::Vector{<:Real}, p::Coulomb)
-#     println("The Coulomb potential has no trainable parameters")
-#     return (dvdnull =  0 .* r ./ d )
-# end
+function grad_potential_energy(r::Vector{<:Real}, p::Coulomb)
+    println("The Coulomb potential has no trainable parameters")
```

Comment:
I'm not familiar with this part of the package, but is calling these functions on a potential that has no trainable parameters valid or not? If it is valid (i.e. the 0 results would be handled gracefully), then the constant `println` would probably pollute the output. If it is not valid, then I would consider changing the `println` to `error` and not return anything.

Reply:
This part of the package needs to be reworked, so I am not terribly concerned with it at the moment (even the format that the methods return is dubious). I will replace this with an error for now.